### ISSUE TITLE
Crash when navigating + searching in Palettes

### ISSUE THREAD

**Issue by 641i130** [musescore/MuseScore#16849]
### Issue type

Crash or freeze

### Bug description

When navigating and attempting to type in the Search (by the Add Palettes tab), the program will crash randomly (after some time after use).

### Steps to reproduce

Just use the program and try typing in the search bar. It will eventually crash.

### MuseScore Version

4.0.2

### Regression

No.

### Operating system

Arch Linux

### Additional context

```
[1336717:1336717:20230315,174319.018171:WARNING thread_snapshot_linux.cc:111] Unknown scheduling policy 1073741825
[1336717:1336717:20230315,174319.022402:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.022753:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.022761:ERROR process_reader_linux.cc:520] no module mappings 0x7f12950ad000
[1336717:1336717:20230315,174319.022812:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.022818:ERROR process_reader_linux.cc:520] no module mappings 0x7f12950a6000
[1336717:1336717:20230315,174319.022930:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.022937:ERROR process_reader_linux.cc:520] no module mappings 0x7f129509e000
[1336717:1336717:20230315,174319.023040:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023047:ERROR process_reader_linux.cc:520] no module mappings 0x7f1295098000
[1336717:1336717:20230315,174319.023092:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023098:ERROR process_reader_linux.cc:520] no module mappings 0x7f129508f000
[1336717:1336717:20230315,174319.023142:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023148:ERROR process_reader_linux.cc:520] no module mappings 0x7f1294865000
[1336717:1336717:20230315,174319.023192:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023198:ERROR process_reader_linux.cc:520] no module mappings 0x7f129485b000
[1336717:1336717:20230315,174319.023242:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023247:ERROR process_reader_linux.cc:520] no module mappings 0x7f129484b000
[1336717:1336717:20230315,174319.023290:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023296:ERROR process_reader_linux.cc:520] no module mappings 0x7f12945ff000
[1336717:1336717:20230315,174319.023339:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023345:ERROR process_reader_linux.cc:520] no module mappings 0x7f12945f3000
[1336717:1336717:20230315,174319.023388:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023393:ERROR process_reader_linux.cc:520] no module mappings 0x7f1294845000
[1336717:1336717:20230315,174319.023436:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023442:ERROR process_reader_linux.cc:520] no module mappings 0x7f1293ffb000
[1336717:1336717:20230315,174319.023484:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023490:ERROR process_reader_linux.cc:520] no module mappings 0x7f1293fdb000
[1336717:1336717:20230315,174319.023533:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023538:ERROR process_reader_linux.cc:520] no module mappings 0x7f12945e7000
[1336717:1336717:20230315,174319.023581:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023585:ERROR process_reader_linux.cc:520] no module mappings 0x7f12932bc000
[1336717:1336717:20230315,174319.023936:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.023942:ERROR process_reader_linux.cc:520] no module mappings 0x7f1292997000
[1336717:1336717:20230315,174319.024101:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.024107:ERROR process_reader_linux.cc:520] no module mappings 0x7f1291214000
[1336717:1336717:20230315,174319.024148:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.024153:ERROR process_reader_linux.cc:520] no module mappings 0x7f12916be000
[1336717:1336717:20230315,174319.024194:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.024199:ERROR process_reader_linux.cc:520] no module mappings 0x7f1291694000
[1336717:1336717:20230315,174319.024589:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.026561:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.026599:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.026673:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.026691:ERROR elf_image_reader.cc:116] no preferred header address
[1336717:1336717:20230315,174319.026693:ERROR process_reader_linux.cc:520] no module mappings 0x7f128c882000
[1336717:1336717:20230315,174319.027199:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027266:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027362:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027395:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027428:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027461:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027495:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027525:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027557:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027586:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027618:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027648:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027702:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027730:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027759:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027789:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027817:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027846:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027876:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027904:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027932:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027961:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.027990:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028018:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028047:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028078:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028323:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028351:ERROR elf_dynamic_array_reader.h:61] tag not found
[1336717:1336717:20230315,174319.028380:ERROR elf_dynamic_array_reader.h:61] tag not found
/tmp/.mount_muse4wAN3ve/AppRun: line 26: 1336702 Segmentation fault      (core dumped) "${APPDIR}/bin/mscore4portable" "$@"
```

**Comment by cbjeukendrup** [comment]
I don't think it is. The output looks similar, but this output is printed for every crash (it seems to come from https://chromium.googlesource.com/chromium/src/third_party/+/master/crashpad/crashpad/snapshot/elf/elf_dynamic_array_reader.h?autodive=0%2F%2F%2F).

**Comment by lyrra** [comment]
Can reproduce, if running under gdb, type in around 20 chars (just hold down a keyboard-key), select all and press any key, which should replace all marked text. Crashes sometimes.

<details> <summary> backtrace </summary>

```
Thread 1 "mscore" received signal SIGSEGV, Segmentation fault.
0x00007ffff5baa1a2 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
(gdb) ba
#0  0x00007ffff5baa1a2 in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#1  0x00007ffff5d8ce34 in QCoreApplicationPrivate::lockThreadPostEventList(QObject*) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#2  0x00007ffff5d8df19 in QCoreApplication::postEvent(QObject*, QEvent*, int) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#3  0x000055555684263f in mu::uicomponents::PopupView::~PopupView() (this=0x5555635a7a60, __in_chrg=<optimized out>) at src/framework/uicomponents/view/popupview.cpp:76
#4  0x000055555686b622 in QQmlPrivate::QQmlElement<mu::uicomponents::PopupView>::~QQmlElement() (this=0x5555635a7a60, __in_chrg=<optimized out>)
    at /usr/include/x86_64-linux-gnu/qt5/QtQml/qqmlprivate.h:144
#5  0x000055555686b64a in QQmlPrivate::QQmlElement<mu::uicomponents::PopupView>::~QQmlElement() (this=0x5555635a7a60, __in_chrg=<optimized out>)
    at /usr/include/x86_64-linux-gnu/qt5/QtQml/qqmlprivate.h:144
#6  0x00007ffff76fd406 in QQmlObjectCreator::clear() () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#7  0x00007ffff7697ca8 in QQmlIncubatorPrivate::clear() () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#8  0x00007ffff7697d79 in QQmlIncubator::clear() () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#9  0x00007ffff53fab88 in  () at /usr/lib/x86_64-linux-gnu/libQt5QmlModels.so.5
#10 0x00007ffff5400281 in  () at /usr/lib/x86_64-linux-gnu/libQt5QmlModels.so.5
#11 0x00007ffff54007eb in  () at /usr/lib/x86_64-linux-gnu/libQt5QmlModels.so.5
#12 0x00007ffff5402139 in QQmlDelegateModel::_q_itemsRemoved(int, int) () at /usr/lib/x86_64-linux-gnu/libQt5QmlModels.so.5
#13 0x00007ffff5404c83 in QQmlDelegateModel::qt_metacall(QMetaObject::Call, int, void**) () at /usr/lib/x86_64-linux-gnu/libQt5QmlModels.so.5
#14 0x00007ffff5dc12f0 in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#15 0x00007ffff5d25ede in QAbstractItemModel::rowsRemoved(QModelIndex const&, int, int, QAbstractItemModel::QPrivateSignal) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#16 0x00007ffff5d2e6ee in QAbstractItemModel::endRemoveRows() () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#17 0x00007ffff5d5ed15 in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#18 0x00007ffff5d5f24c in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#19 0x00007ffff5d5f639 in  () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#20 0x00007ffff5d60336 in QSortFilterProxyModel::setFilterFixedString(QString const&) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#21 0x00007ffff5d631f3 in QSortFilterProxyModel::qt_metacall(QMetaObject::Call, int, void**) () at /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#22 0x00005555565f4f78 in mu::palette::PaletteCellFilterProxyModel::qt_metacall(QMetaObject::Call, int, void**)
    (this=0x55555a7d1b00, _c=QMetaObject::InvokeMetaMethod, _id=74, _a=0x7fffffff3360)
    at builds/Linux-Qt-usr-Make-Debug/src/palette/palette_autogen/OTUN7GXZT4/moc_palettemodel.cpp:481
#23 0x00007ffff76ce6fd in  () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#24 0x00007ffff75b7b0e in  () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#25 0x00007ffff75b995a in QV4::QObjectMethod::callInternal(QV4::Value const*, QV4::Value const*, int) const () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#26 0x00007ffff75e66ca in QV4::Runtime::CallPropertyLookup::call(QV4::ExecutionEngine*, QV4::Value const&, unsigned int, QV4::Value*, int) () at /usr/lib/x86_64-linux-gnu/libQt5Qml.so.5
#27 0x00007fffb2b21a76 in  ()

```

</details>

**Comment by icactus** [comment]
1. Create new score with any instrument.
2. Add several notes and select them (If only one note or an empty measure is selected I do not run into this problem).
3. In palettes sidebar click on the search magnifying glass.
4. Quickly type a word that requires the menu to auto-populate multiple results (like "accent" where it will also begin to match "accidental")
5. Only some of the typed letters will register and the program will freeze and crash shortly thereafter.

If I type very slowly there is no problem.

OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.2-230651553, revision: github-musescore-musescore-dbe7c6d

**Comment by adazem009** [comment]
I think the description of this issue should be changed. I can reproduce this in earlier versions of MuseScore 4.x (4.0.0 and 4.0.1).

**Comment by 641i130** [comment]
@adazem009 how should I update it?

**Comment by adazem009** [comment]
@641i130 Change the regression to "No." because it didn't work in previous versions of MuseScore. Also remove the "regression" label from the issue.

**Comment by DmitryArefiev** [comment]
Fixed in #17864
